### PR TITLE
Add CI/CD with GitHub Actions

### LINKED ISSUE #305: Reestablish CI/CD with GitHub Actions

### PR DESCRIPTION
**Pull request by jginsburgn**
Fixes #305

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  linux:
+    name: "Node ${{ matrix.node }} on Linux: Test and Lint"
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node:
+          - 10
+          - 12
+    steps:
+      - uses: actions/checkout@v2
```
Comment (XhmikosR) [inline .github/workflows/test.yml:18]:
```suggestion
      - uses: actions/checkout@v3
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        with:
+          fetch-depth: 0
+      - uses: browser-actions/setup-firefox@latest
+      - uses: actions/setup-node@v2
```
Comment (XhmikosR) [inline .github/workflows/test.yml:22]:
```suggestion
      - uses: actions/setup-node@v3
```

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    defaults:
+      run:
+        shell: bash
+    strategy:
+      matrix:
+        node:
+          - 10
+          - 12
+    steps:
+      - uses: actions/checkout@v2
```
Comment (XhmikosR) [inline .github/workflows/test.yml:40]:
```suggestion
      - uses: actions/checkout@v3
```

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+      - uses: actions/setup-node@v2
```
Comment (XhmikosR) [inline .github/workflows/test.yml:41]:
```suggestion
      - uses: actions/setup-node@v3
```

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+jobs:
+  linux:
+    name: "Node ${{ matrix.node }} on Linux: Test and Lint"
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node:
+          - 10
+          - 12
```
Comment (XhmikosR) [inline .github/workflows/test.yml:16]: Should also test 14 and 16 IMHO :)

Comment (jginsburgn) [reply .github/workflows/test.yml:16]: Agreed!

File: .gitignore [hunk 6]
```diff
@@ -3,3 +3,4 @@ node_modules/
 tmp/
 lib/adapter.js
 integration-tests/
+.DS_Store
```
Comment (XhmikosR) [inline .gitignore:6]: Missing EOF newline here and in package.json.

File: .github/workflows/test.yml [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  linux:
+    name: "Node ${{ matrix.node }} on Linux: Test and Lint"
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
```
Comment (XhmikosR) [inline .github/workflows/test.yml:13]: All good now, just add a `fail-fast: false` in each `strategy` to match the previous behavior

File: spec/.eslintrc [hunk 8]
```diff
@@ -0,0 +1,5 @@
+{
+  "env": {
+    "jasmine": true
+  }
+}
```
Comment (mtrea) [inline spec/.eslintrc:5]: Please add an EOF newline

**Review by mtrea — APPROVED** [review]

**Review by mtrea — APPROVED** [review]

**Comment by XhmikosR** [comment]
Note that eslint 8.x isn't compatible with all the eslint* packages right now.

**Pull request merged**